spec/patterns/bind_spec.rb

```diff
@@ -0,0 +1,57 @@
+# encoding: utf-8
+require "spec_helper"
+require "logstash/patterns/core"
+
+describe_pattern "BIND9", ['legacy', 'ecs-v1'] do
+
+  let(:message) do # Bind 9.10
+    '17-Feb-2018 23:06:56.326 queries: info: client 172.26.0.1#12345 (test.example.com): query: test.example.com IN A +E(0)K (172.26.0.3)'
+  end
+
+  it 'matches' do
+    should include("timestamp" => "17-Feb-2018 23:06:56.326")
+    if ecs_compatibility?
+      should include("log" => hash_including("level" => "info"))
+      should include("client" => { "ip" => "172.26.0.1", "port" => 12345 })
```

Comment:
guess, yet again, these should be `source.ip` and `source.port` instead of `client.*`

Reply:
In ECS we try to have folks always use source/destination as the baseline, and only populate client/server when it makes sense.

However DNS is one of the examples where client/server makes the most sense, because it's UDP.

If looking at a stream of activity with questions and answers, using source/dest you'd get a hard to follow view like:

source.ip | destination.ip | dns.type | question.name
--|--|--|--
10.42.42.42 | 1.1.1.1 | query | google.com
1.1.1.1 | 10.42.42.42 | answer | google.com

Whereas if we view by client/server we have a more consistent view:

client.ip | server.ip | dns.type | question.name
--|--|--|--
10.42.42.42 | 1.1.1.1 | query | google.com
10.42.42.42 | 1.1.1.1 | answer | google.com


So for that reason I would definitely agree with going directly for client/server in the groks, as you're already doing.

Note: I think bind doesn't give us answers, but I would go with client/server nonetheless. It will make it easier to correlate with other sources of DNS events also populating client/server like Packetbeat, Suricata, Zeek.